### DISCUSSION TITLE
There is no permission named `view-organizations` for guard `web`.

### DISCUSSION THREAD

**Discussion by loxK** [spatie/laravel-permission#2471] [Q&A] [answered]
Hello,

I am developing my first **Laravel** app. I am using `Breeze` with a separated `Vue PWA` frontend. I need to add permissions to my app and this simple code, in a database `Seeder`, throws an error :

```php

// Reset cached roles and permissions
app()[PermissionRegistrar::class]->forgetCachedPermissions();

// create permissions
Permission::create(['name' => 'view-organizations']);
$role2 = Role::create(['name' => 'professional']);
$role2->syncPermissions([
    'view-organizations',
]);
```

```
 There is no permission named `view-organizations` for guard `web`.

  at vendor/spatie/laravel-permission/src/Exceptions/PermissionDoesNotExist.php:11
  
  8   database/seeders/Init.php:31
      Spatie\Permission\Models\Role::syncPermissions()
      +36 vendor frames 

  45  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()
```

I have followed it to `Spatie\Permission\Models:findByName` function, but I am then stuck, not quite understanding what is going wrong. Any idea of what could cause that issue ?

**Comment by loxK** [comment, marked as answer]
I hate finding the solution after hours searching and asking for help.

But here it is. I am using `WithoutModelEvents` trait in my `Seeder`, so I need to call `app()[PermissionRegistrar::class]->forgetCachedPermissions()` after permissions are seeded as per https://github.com/spatie/laravel-permission/issues/2342#issuecomment-1451453961

**Reply by drbyte** [reply]
Thanks for posting back with your solution. I've updated the docs so it's easier to find.
